compiler/rustc_lint/messages.ftl

```diff
@@ -241,8 +241,20 @@ lint_hidden_unicode_codepoints = unicode codepoint changing visible direction of
 lint_identifier_non_ascii_char = identifier contains non-ASCII characters
 
 lint_identifier_uncommon_codepoints = identifier contains {$codepoints_len ->
-    [one] an uncommon Unicode codepoint
-    *[other] uncommon Unicode codepoints
+    [one] { $identifier_type ->
+        [Exclusion] an {$identifier_type} Unicode codepoint
+        [Technical] a {$identifier_type} Unicode codepoint
+        [Limited_Use] a {$identifier_type} Unicode codepoint
+        [Not_NFKC] a {$identifier_type} Unicode codepoint
+        *[other] an uncommon Unicode codepoint
+    }
+    *[other] { $identifier_type ->
+        [Exclusion] {$identifier_type} Unicode codepoints
+        [Technical] {$identifier_type} Unicode codepoints
```

Comment:
I feel like all these variants could use a user friendly name

Reply:
+1. I would use a more down-to-earth description (I really like Manish's short definitions from #120228) in the diagnostic message and put technically name in a diagnostic note.

Reply:
Indeed, I planned on adding `note`s appending each of the diagnostics explaining each of the categories of codepoints.